src/main/java/io/gravitee/common/util/VertxProxyOptionsUtils.java

```diff
@@ -32,31 +32,34 @@ public class VertxProxyOptionsUtils {
     static final String PROXY_USERNAME_PROPERTY = "system.proxy.username";
     static final String PROXY_PASSWORD_PROPERTY = "system.proxy.password";
 
-    public static ProxyOptions initFromSpringContext(ApplicationContext applicationContext) {
-        final Environment environment = applicationContext.getEnvironment();
+    public static void setSystemProxy(HttpClientOptions options, Configuration configuration) {
+        options.setProxyOptions(buildProxyOptions(configuration));
+    }
+
+    public static ProxyOptions buildProxyOptions(Configuration configuration) {
```

Comment:
@brasseld In order to keep gateway [endpoint resolvers](https://github.com/gravitee-io/gravitee-api-management/pull/1784/files) able to build proxy options once and for all, maybe we should keep this method and propose the other as an alternative ?